### PR TITLE
SRCH-5312 Improve error logging for search-gov

### PR DESCRIPTION
**Pull request by krbhavith**
## Summary
- Send whole exception instead of just the error message in the logger.
 
### Checklist
Please ensure you have addressed all concerns below before marking a PR "ready for review" or before requesting a re-review. If you cannot complete an item below, replace the checkbox with the ⚠️ `:warning:` emoji and explain why the step was not completed.
 
#### Functionality Checks
 
- [x] You have merged the latest changes from the target branch (usually `main`) into your branch.
  
- [x] Your primary commit message is of the format **SRCH-#### \<description\>** matching the associated Jira ticket.

- [x] PR title is either of the format **SRCH-#### \<description\>** matching the associated Jira ticket (i.e. "SRCH-123 implement feature X"), or **Release - SRCH-####, SRCH-####, SRCH-####** matching the Jira ticket numbers in the release.
 
- [x] Automated checks pass. If Code Climate checks do not pass, explain reason for failures:
 
#### Process Checks

- [x] You have specified at least one "Reviewer".

### CODE REVIEW

**Review by stevenbarragan — CHANGES_REQUESTED** [review]
We need to send the exception directly to the logger not just in a string.

File: app/jobs/affiliate_indexed_document_fetcher_job.rb [hunk 1]
```diff
@@ -9,9 +9,9 @@ def perform(affiliate_id, start_id, end_id, scope)
       where(conditions).order(:last_crawled_at).each do |indexed_document|
       IndexedDocument.find(indexed_document.id).fetch
     rescue ActiveRecord::RecordNotFound => e
-      Rails.logger.warn "Cannot find IndexedDocument to fetch: #{e}"
+      Rails.logger.warn 'Cannot find IndexedDocument to fetch:', e
     end
   rescue ActiveRecord::RecordNotFound => e
-    Rails.logger.warn "Ignoring race condition in AffiliateIndexedDocumentFetcherJob: #{e}"
+    Rails.logger.warn 'Ignoring race condition in AffiliateIndexedDocumentFetcherJob:', e
```
Comment (stevenbarragan) [inline app/jobs/affiliate_indexed_document_fetcher_job.rb:15]: Can we change this to log the exception as `error` instead?
```
 Rails.logger.error 'Ignoring race condition in AffiliateIndexedDocumentFetcherJob:', e
 ```

File: lib/geoip_lookup.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@ def self.establish_connection!
   def self.lookup(ip)
     @geoip_db.city(ip)
   rescue SocketError => e
-    Rails.logger.warn("GeoipLookup failed lookup on #{ip}: #{e.message}")
+    Rails.logger.warn "GeoipLookup failed lookup on #{ip}", e
```
Comment (stevenbarragan) [inline lib/geoip_lookup.rb:9]: Can we use `Rails.logger.warn` here instead?

**Review by stevenbarragan — CHANGES_REQUESTED** [review]
One last change.

File: lib/geoip_lookup.rb (hunk 2 above)
Comment (krbhavith) [reply lib/geoip_lookup.rb:9]: You mean `Rails.logger.error`?

**Review by stevenbarragan — APPROVED** [review]
LGTM

**Pull request merged**